app/screens/more_dms/index.js

```diff
@@ -3,39 +3,68 @@
 
 import {bindActionCreators} from 'redux';
 import {connect} from 'react-redux';
+import {createSelector} from 'reselect';
 
 import {setChannelDisplayName} from 'app/actions/views/channel';
 import {makeDirectChannel} from 'app/actions/views/more_dms';
 import {getTheme} from 'app/selectors/preferences';
-import {getProfiles, searchProfiles} from 'mattermost-redux/actions/users';
+
+import {getProfiles, getProfilesInTeam, searchProfiles} from 'mattermost-redux/actions/users';
+import {General} from 'mattermost-redux/constants';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
 import {getMyPreferences} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentTeamId} from 'mattermost-redux/selectors/entities/teams';
+import {getCurrentUserId, getProfilesInCurrentTeam, getUsers} from 'mattermost-redux/selectors/entities/users';
 
 import MoreDirectMessages from './more_dms';
 
+function sortAndRemoveCurrentUser(profiles, currentUserId) {
+    const users = {...profiles};
+    Reflect.deleteProperty(users, currentUserId);
+    return Object.values(users).sort((a, b) => {
+        const nameA = a.username;
+        const nameB = b.username;
+
+        return nameA.localeCompare(nameB);
+    });
+}
+
+const getUsersInCurrentTeamForMoreDirectMessages = createSelector(
```

Comment:
The `FilteredList` works a bit differently since it manually removes the current user and sorts the users itself. And it could probably be used in the web app, but that's not guaranteed